chain/client/src/sync.rs

```diff
@@ -768,11 +771,24 @@ impl StateSync {
                     }
                 }
                 ShardSyncStatus::StateDownloadComplete => {
-                    this_done = true;
                     let shard_state_header = chain.get_state_header(shard_id, sync_hash)?;
                     let state_num_parts =
                         get_num_state_parts(shard_state_header.state_root_node().memory_usage);
                     chain.clear_downloaded_parts(shard_id, sync_hash, state_num_parts)?;
+                    if split_states {
+                        *shard_sync_download = ShardSyncDownload {
+                            downloads: vec![],
+                            status: ShardSyncStatus::StateSplit,
+                        }
+                    } else {
+                        this_done = true;
+                    }
+                }
+                ShardSyncStatus::StateSplit => {
+                    assert!(split_states);
```

Comment:
It is a local check that checks the implementation of the code that triggers StateSplit is correct. I'll change it to debug_assert so that this assert will not be triggered at release code